test/datatype/unpack_ooo.c

```diff
@@ -151,7 +151,8 @@ static int testcase(ompi_datatype_t * newtype, size_t arr[][2]) {
                     displ = (char*)&bar[j] - (char*)&bar[0];
                 }
                 for (i = 0; 0 != arr[i][0]; i++) {
-                    if( (displ >= arr[i][1]) && (displ <= (arr[i][1] + arr[i][0])) ) {
+                    if( (displ >= (ptrdiff_t) arr[i][1]) &&
```

Comment:
displ is a ptrdiff_t already because of a compiler complaint, (because it is computed as a difference between two pointers, while in fact is the displacement of a field in a struct).

Reply:
Here's the compiler warnings it generated without updating the RHS to be `(ptrdiff_t)`:

```
1024 unpack_ooo.c: In function ‘testcase’:
1025 unpack_ooo.c:154:32: warning: comparison of integer expressions of different \
     signedness: ‘ptrdiff_t’ {aka ‘long int’} and ‘size_t’ {aka ‘long unsigned int\
     ’} [-Wsign-compare]
1026                      if( (displ >= arr[i][1]) && (displ <= (arr[i][1] + arr[i\
     ][0])) ) {
1027                                 ^~
1028 unpack_ooo.c:154:56: warning: comparison of integer expressions of different \
     signedness: ‘ptrdiff_t’ {aka ‘long int’} and ‘size_t’ {aka ‘long unsigned int\
     ’} [-Wsign-compare]
1029                      if( (displ >= arr[i][1]) && (displ <= (arr[i][1] + arr[i\
     ][0])) ) {
1030                                                         ^~
```

I don't have an opinion on this one; do you prefer a different fix?

Reply:
Based on a machine understanding of a piece of code. I don't really care.